openstack/CleanWlmDatabase/historical_data_cleanup/DeleteHistoricalRecords.py

```diff
@@ -0,0 +1,88 @@
+import os
+import configparser
+import re
+import mysql.connector
+
+# Get records to be deleted using select query
+# Write those records into file
+# Run delete on those records
+
+batchSize=1000
+
+# DB connection 
+## Grab fields from the conf file
+config = configparser.ConfigParser()
+config.read('/etc/workloadmgr/workloadmgr.conf')
+mysql_conn = config['DEFAULT']['sql_connection']
+db_string = re.split('mysql://|=|@|/|:|\?', mysql_conn)
+
+## Edit these 4 variable manually if you want to change any of them
+db_user = db_string[1]
+db_pass = db_string[2]
+db_host = db_string[3]
+db_db = db_string[4]
+
+my_connect = mysql.connector.connect(user=db_user, password=db_pass, host=db_host, database=db_db)
+cursor = my_connect.cursor(dictionary=True)
+
+
+def getRecords(batchSize):
+    try:
+       filepath="./select_queries/{}".format(table)
+       tablename = open(filepath,"r")
+       #print(tablename)
+       query=tablename.readline()
+       tablename.close()
+       record_count=1
+       while record_count > 0:
+          print('\n\nGetting records from {}'.format(table))
+          query=query.replace("#LIMIT#",str(batchSize))
+          print("Running query : {}".format(query))
+          cursor.execute(query)
```

Comment:
The select queries have joins which would be running on tables have huge amount of data. It might cause a timeout of the query. What would happen in such case? Is that handled?

Reply:
In Case of timeout it will be catched in exception 
and in case of exception it will again fetch records from database

Reply:
So if select times out after say 1 hour, then what would be the next step?